### PR TITLE
feat: added readme docs for subscription feature

### PR DESCRIPTION
**Pull request by mrazadar**
This PR  remove new relic logs previously added while testing 3DS issue on prod. 

Ticket: https://2u-internal.atlassian.net/browse/PON-453

### CODE REVIEW

**Review by NawfalAhmed — APPROVED** [review]

File: README.rst [hunk 1]
```diff
@@ -229,3 +229,30 @@ Appendix B: Adding No-Op Stuff to Test Sandbox Deploys
 ----------------------------------------------------------
 
 Let's try this.
+
+
+Subscriptions
+-------------
+
+This payment MFE has introduced an additional React route, ``payment.stage.edx.org/subscription``, specifically designed to handle recurring payments. Currently, we are only permitting one payment method to manage recurring purchases, namely stripe subscriptions.
```
Comment (julianajlk) [inline README.rst:237]: Can you specify `handle B2C recurring payments`

File: README.rst [hunk 2]
```diff
@@ -229,3 +229,30 @@ Appendix B: Adding No-Op Stuff to Test Sandbox Deploys
...
+
+Development
+-----------
+
+All functionality related to subscriptions is built under the ``src/subscription`` directory. It seeks to optimize the utilization of existing functionality to implement this new feature, adding any necessary enhancements on top of it. The following list outlines key points where it diverges from the ``src/payment`` flow:
+
+- Main ``src/index.scss`` includes ``src/subscription/index.scss`` after ``src/payment/index.scss`` allowing it to override the payment page styles.
```
Comment (julianajlk) [inline README.rst:244]: Great info to know, thanks for adding!

File: README.rst [hunk 3]
```diff
@@ -229,3 +229,30 @@ Appendix B: Adding No-Op Stuff to Test Sandbox Deploys
...
+- ``src/subscription/i18n-protected-messages`` overrides the *Transifex* messages with the locally (hardcoded) defined messages, specifically for ``es-419`` locale.
+- ``src/subscription/test-utils`` defines the new test wrapper with ``react-testing-library``. We use ``react-testing-library`` to write unit tests for ``src/subscription`` code.
+- ``src/subscription/alerts`` defines its new set of alerts specific to ``Subscription`` page.
+- ``SUBSCRIPTIONS_BASE_URL`` exposes different endpoints for ``GET`` and ``POST`` subscription CTAs.
```
Comment (julianajlk) [inline README.rst:248]: Suggestion: `exposes different subscriptions backend endpoints`, it's implied but just a reminder these refer to our subscriptions BE repo

File: README.rst [hunk 4]
```diff
@@ -229,3 +229,30 @@ Appendix B: Adding No-Op Stuff to Test Sandbox Deploys
...
+
+3DS
+---
+
+For recurring payments, we have also enabled the 3D Secure payment flow. All the basic functionality for 3DS is implemented under the ``src/subscription/secure-3d`` directory. The following key points outline the 3DS implementation:
+
+- We render 3DS bank details inside an ``iframe`` within a modal. . The user completes the 3DS check, and the bank redirects the user back to our provided return address, i.e., ``payment.stage.edx.org/subscription/3ds``.
```
Comment (julianajlk) [inline README.rst:255]: Double `.  . ` intentional?

**Review by julianajlk — APPROVED** [review]
Thank you for adding this documentation!

**Review by MuhammadMattiullah — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (mrazadar) [reply README.rst:237]: https://2u-internal.atlassian.net/wiki/spaces/RS/pages/551321788/How+to+handle+b2c+recurring+payments+on+the+payment+checkout+page

File: README.rst (hunk 3 above)
Comment (mrazadar) [reply README.rst:248]: done. https://2u-internal.atlassian.net/wiki/spaces/RS/pages/551321788/How+to+handle+b2c+recurring+payments+on+the+payment+checkout+page

File: README.rst (hunk 4 above)
Comment (mrazadar) [reply README.rst:255]: done.

**Pull request merged**